### DISCUSSION TITLE
How to initialize qubits in a specific state?

### DISCUSSION THREAD

**Discussion by ellow0rld** [NVIDIA/cuda-quantum#1747] [Q&A]
I am trying to initialize qubits in a particular state. I noticed that there's a [github issue](https://github.com/NVIDIA/cuda-quantum/issues/1086) discussing the same problem. I tried implementing the solution provided there. The function runs without errors when it's not called, but I encounter errors when I call it.

```
c = [.70710678, 0., 0., 0.70710678]
@cudaq.kernel
def test():
    q = cudaq.qvector(c)
```

No error when I run the above

```
c = [.70710678, 0., 0., 0.70710678]
@cudaq.kernel
def test():
    q = cudaq.qvector(c)

test()
```

But, when I call it, I get the following error
> error: 'quake.alloca' op operand #0 must be signless integer, but got '!cc.stdvec<f64>'

I don't know why it's not working. Is it a version problem?

**Specifications**
I am running cudaq on VSCode, installed using Docker.

`nvidia-smi`
![image](https://github.com/NVIDIA/cuda-quantum/assets/116413038/b5a82bd9-e97b-4de8-99c2-71353fd8566d)

**Comment by bmhowe23** [comment]
Hi @ellow0rld - since https://github.com/NVIDIA/cuda-quantum/issues/1086 will take many incremental updates to implement, the development is not being done in the main branch. Rather, it is being done in the https://github.com/NVIDIA/cuda-quantum/tree/experimental/stateHandling branch. You can experiment with this branch yourself by using the nvcr.io/nvidia/nightly/cuda-quantum:experimental-stateHandling-base Docker image. However, note that features may change quickly in that branch, and as such, it should be considered - in some sense - less stable than the main branch.

**Reply by ellow0rld** [reply]
Hi @bmhowe23 Thank you so much for the response. Will try experimenting the stateHandling branch!!

**Reply by efratshabtai** [reply]
@ellow0rld the state handling branch have been merged to main and released as part of 0.8.0 version, please try that and let us know if we can close this discussion.
